src/wallet/spend.cpp

```diff
@@ -382,6 +382,15 @@ bool CWallet::AttemptSelection(const CAmount& nTargetValue, const CoinEligibilit
         results.emplace_back(std::make_tuple(waste, std::move(knapsack_coins), knapsack_value));
     }
 
+    // We include the minimum final change for SRD as we do want to avoid making really small change.
+    // KnapsackSolver does not need this because it includes MIN_CHANGE internally.
+    const CAmount srd_target = nTargetValue + coin_selection_params.m_change_fee + MIN_FINAL_CHANGE;
```

Comment:
any simulations/intuitions on what these change outputs look like vs knapsack values, aside from the difference in constant?

Why was this constant `MIN_FINAL_CHANGE` defined but never used before? How was it picked?

Reply:
It looks like we lost it during the effective value change. `MIN_FINAL_CHANGE` used to be the minimum allowed change output with the looping behavior.